src/test/java/org/opensearch/knn/TestUtils.java

```diff
@@ -32,6 +30,62 @@
     public static final String BWCSUITE_ROUND = "tests.rest.bwcsuite_round";
     public static final String TEST_CLUSTER_NAME = "tests.clustername";
 
+    public static float[][] generateVectors(int numVectors, int dimensions) {
+        float[][] Vectors = new float[numVectors][dimensions];
+        int p = 50, q = 10;
+
+        for (int i = 0; i < numVectors; i++) {
+            float[] vector = new float[dimensions];
+            for (int j = 0; j < dimensions; j++) {
+                vector[j] = (float) ((j * j * p++) - (i * i * q--));
+            }
+            Vectors[i] = vector;
+        }
+        return Vectors;
+    }
+
+    public static float[][] generateQueryVectors(int numVectors, int dimensions){
+        float[][] Vectors = new float[numVectors][dimensions];
+        int m = 1000, n = 250;
+        for (int i = 0; i < numVectors; i++) {
+            float[] vector = new float[dimensions];
+            for (int j = 0; j < dimensions; j++) {
+                vector[j] = (float)((i * m++) - (j * n--));
+                m=m+5;
+                n=n+7;
+            }
+            Vectors[i] = vector;
+        }
+        return Vectors;
+    }
+
+    public static float[][] generateRandomVectors(int numVectors, int dimensions){
+        float[][] Vectors = new float[numVectors][dimensions];
+
+        for (int i = 0; i < numVectors; i++) {
+            float[] vector = new float[dimensions];
+            for (int j = 0; j < dimensions; j++) {
+                vector[j] = random().nextFloat();
+            }
+            Vectors[i] = vector;
+        }
+        return Vectors;
+    }
+
+    public static float[][] computeGroundTruthDistances(float[][] indexVectors, float[][] queryVectors, SpaceType spaceType, int k){
+        float[][] groundTruthValues = new float[queryVectors.length][k];
```

Comment:
This function could use a long comment about what it is doing.

Reply:
added the comment